--- Qt/ApplicationComponents/Resources/UI/pqExampleVisualizationsDialog.ui	2024-08-21 20:55:47.000000000 +0200
+++ Qt/ApplicationComponents/Resources/UI/pqExampleVisualizationsDialog.ui	2024-09-10 09:33:13.133348346 +0200
@@ -8,55 +8,22 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>682</width>
-    <height>666</height>
+    <width>996</width>
+    <height>420</height>
    </rect>
   </property>
   <property name="windowTitle">
    <string>ParaView Example Visualizations</string>
   </property>
   <layout class="QGridLayout" name="gridLayout">
-   <item row="1" column="0" colspan="2">
+   <item row="1" column="0" colspan="3">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Click on one of thumbnails below to load an example visualization&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </item>
-   <item row="4" column="0">
-    <widget class="QPushButton" name="HotGasAnalysisExampleButton">
-     <property name="text">
-      <string/>
-     </property>
-     <property name="icon">
-      <iconset resource="../pqApplicationComponents.qrc">
-       <normaloff>:/pqApplicationComponents/Thumbnails/HotGasAnalysisExample.png</normaloff>:/pqApplicationComponents/Thumbnails/HotGasAnalysisExample.png</iconset>
-     </property>
-     <property name="iconSize">
-      <size>
-       <width>300</width>
-       <height>225</height>
-      </size>
-     </property>
-     <property name="flat">
-      <bool>true</bool>
-     </property>
-    </widget>
-   </item>
-   <item row="5" column="0">
-    <widget class="QLabel" name="HotGasAnalysisExampleDescription">
-     <property name="text">
-      <string>Data analysis of hot gas from Exodus II file</string>
-     </property>
-     <property name="alignment">
-      <set>Qt::AlignHCenter|Qt::AlignTop</set>
-     </property>
-     <property name="wordWrap">
-      <bool>true</bool>
-     </property>
-    </widget>
-   </item>
-   <item row="0" column="0" colspan="2">
+   <item row="0" column="0" colspan="3">
     <widget class="QLabel" name="titleLabel">
      <property name="font">
       <font>
@@ -67,43 +34,17 @@
       </font>
      </property>
      <property name="text">
-      <string>ParaView Example Visualizations</string>
+      <string>TTK Example Visualizations</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
-   <item row="5" column="1">
-    <widget class="QLabel" name="WaveletExampleDescription">
-     <property name="text">
-      <string>Wavelet with volume rendering and contours</string>
-     </property>
-     <property name="alignment">
-      <set>Qt::AlignHCenter|Qt::AlignTop</set>
-     </property>
-     <property name="wordWrap">
-      <bool>true</bool>
-     </property>
-    </widget>
-   </item>
-   <item row="8" column="1">
-    <widget class="QLabel" name="HeadSQExampleDescription">
-     <property name="text">
-      <string>Contouring CT scan of a head</string>
-     </property>
-     <property name="alignment">
-      <set>Qt::AlignHCenter|Qt::AlignTop</set>
-     </property>
-     <property name="wordWrap">
-      <bool>true</bool>
-     </property>
-    </widget>
-   </item>
-   <item row="8" column="0">
-    <widget class="QLabel" name="TosExampleDescription">
+   <item row="3" column="0">
+    <widget class="QLabel" name="Example1Description">
      <property name="text">
-      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Sea surface temperatures&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
+      <string>Scalar data example</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignHCenter|Qt::AlignTop</set>
@@ -113,7 +54,7 @@
      </property>
     </widget>
    </item>
-   <item row="9" column="0" colspan="2">
+   <item row="8" column="0" colspan="3">
     <widget class="QDialogButtonBox" name="buttonBox">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
@@ -124,9 +65,9 @@
     </widget>
    </item>
    <item row="3" column="1">
-    <widget class="QLabel" name="DiskOutRefExampleDescription">
+    <widget class="QLabel" name="Example2Description">
      <property name="text">
-      <string>Exodus II file, Clip filter, Stream Tracer filter, Tube filter, Glyph filter</string>
+      <string>Bivariate scalar data example</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignHCenter|Qt::AlignTop</set>
@@ -136,55 +77,27 @@
      </property>
     </widget>
    </item>
-   <item row="4" column="1">
-    <widget class="QPushButton" name="WaveletExampleButton">
+   <item row="3" column="2">
+    <widget class="QLabel" name="Example3Description">
      <property name="text">
-      <string/>
-     </property>
-     <property name="icon">
-      <iconset resource="../pqApplicationComponents.qrc">
-       <normaloff>:/pqApplicationComponents/Thumbnails/WaveletExample.png</normaloff>:/pqApplicationComponents/Thumbnails/WaveletExample.png</iconset>
-     </property>
-     <property name="iconSize">
-      <size>
-       <width>300</width>
-       <height>225</height>
-      </size>
-     </property>
-     <property name="flat">
-      <bool>true</bool>
+      <string>Uncertain data example</string>
      </property>
-    </widget>
-   </item>
-   <item row="7" column="1">
-    <widget class="QPushButton" name="HeadSQExampleButton">
-     <property name="text">
-      <string/>
-     </property>
-     <property name="icon">
-      <iconset resource="../pqApplicationComponents.qrc">
-        <normalon>:/pqApplicationComponents/Thumbnails/HeadSQExample.png</normalon>
-      </iconset>
-     </property>
-     <property name="iconSize">
-      <size>
-       <width>300</width>
-       <height>225</height>
-      </size>
+     <property name="alignment">
+       <set>Qt::AlignHCenter|Qt::AlignTop</set>
      </property>
-     <property name="flat">
+     <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </item>
-   <item row="7" column="0">
-    <widget class="QPushButton" name="TosExampleButton">
+   <item row="2" column="0">
+    <widget class="QPushButton" name="Example1Button">
      <property name="text">
       <string/>
      </property>
      <property name="icon">
-      <iconset resource="../pqApplicationComponents.qrc">
-        <normalon>:/pqApplicationComponents/Thumbnails/TosExample.png</normalon>
+      <iconset>
+        <normalon>:/pqApplicationComponents/Thumbnails/Example1.png</normalon>
       </iconset>
      </property>
      <property name="iconSize">
@@ -193,32 +106,16 @@
        <height>225</height>
       </size>
      </property>
-     <property name="flat">
-      <bool>true</bool>
-     </property>
-    </widget>
-   </item>
-   <item row="3" column="0">
-    <widget class="QLabel" name="CanExampleDescription">
-     <property name="text">
-      <string>Exodus II file with timesteps, Clip filter</string>
-     </property>
-     <property name="alignment">
-      <set>Qt::AlignHCenter|Qt::AlignTop</set>
-     </property>
-     <property name="wordWrap">
-      <bool>true</bool>
-     </property>
     </widget>
    </item>
    <item row="2" column="1">
-    <widget class="QPushButton" name="DiskOutRefExampleButton">
+    <widget class="QPushButton" name="Example2Button">
      <property name="text">
       <string/>
      </property>
      <property name="icon">
       <iconset>
-       <normalon>:/pqApplicationComponents/Thumbnails/DiskOutRefExample.png</normalon>
+       <normalon>:/pqApplicationComponents/Thumbnails/Example2.png</normalon>
       </iconset>
      </property>
      <property name="iconSize">
@@ -232,14 +129,14 @@
      </property>
     </widget>
    </item>
-   <item row="2" column="0">
-    <widget class="QPushButton" name="CanExampleButton">
+   <item row="2" column="2">
+    <widget class="QPushButton" name="Example3Button">
      <property name="text">
       <string/>
      </property>
      <property name="icon">
-      <iconset>
-       <normalon>:/pqApplicationComponents/Thumbnails/CanExample.png</normalon>
+      <iconset >
+       <normalon>:/pqApplicationComponents/Thumbnails/Example3.png</normalon>
       </iconset>
      </property>
      <property name="iconSize">
@@ -269,12 +166,9 @@
   </layout>
  </widget>
  <tabstops>
-  <tabstop>CanExampleButton</tabstop>
-  <tabstop>DiskOutRefExampleButton</tabstop>
-  <tabstop>HeadSQExampleButton</tabstop>
-  <tabstop>HotGasAnalysisExampleButton</tabstop>
-  <tabstop>WaveletExampleButton</tabstop>
-  <tabstop>TosExampleButton</tabstop>
+  <tabstop>Example1Button</tabstop>
+  <tabstop>Example2Button</tabstop>
+  <tabstop>Example3Button</tabstop>
   <tabstop>buttonBox</tabstop>
  </tabstops>
  <resources>
