--- a/CMakeLists.txt	2016-07-26 21:52:16.000000000 +0200
+++ b/CMakeLists.txt	2016-09-23 01:43:48.198268659 +0200
@@ -939,3 +939,17 @@
       "paraview-source-${format}")
   endforeach ()
 endif ()
+
+#-----------------------------------------------------------------------------
+# Install TTK example data
+install(FILES ${ParaView_SOURCE_DIR}/TTK/Data/Example1.vti
+  DESTINATION ${VTK_INSTALL_DATA_DIR}/data/
+  COMPONENT Development)
+
+install(FILES ${ParaView_SOURCE_DIR}/TTK/Data/Example2.vti
+  DESTINATION ${VTK_INSTALL_DATA_DIR}/data/
+  COMPONENT Development)
+
+install(FILES ${ParaView_SOURCE_DIR}/TTK/Data/Example3.vti
+  DESTINATION ${VTK_INSTALL_DATA_DIR}/data/
+  COMPONENT Development)
