diff --git a/Clients/ParaView/CMakeLists.txt b/Clients/ParaView/CMakeLists.txt
index f51fe395b..3044bfe08 100644
--- a/Clients/ParaView/CMakeLists.txt
+++ b/Clients/ParaView/CMakeLists.txt
@@ -16,9 +16,9 @@ if (PARAVIEW_ENABLE_EMBEDDED_DOCUMENTATION)
 endif ()

 if (DEFINED PARAVIEW_VERSION_NICKNAME)
-  set(paraview_version "${PARAVIEW_VERSION_NICKNAME}")
+  set(paraview_version "[TTK 1.2] ${PARAVIEW_VERSION_NICKNAME}")
 else ()
-  set(paraview_version "${PARAVIEW_VERSION_FULL}")
+  set(paraview_version "[TTK 1.2] ${PARAVIEW_VERSION_FULL}")
 endif ()
 set(paraview_title "ParaView ${paraview_version}")
 if (PARAVIEW_BRANCH)
@@ -65,7 +65,7 @@ paraview_client_add(
   EXPORT            "ParaViewClient"
   APPLICATION_NAME  "ParaView"
   TITLE             "${paraview_title}"
-  ORGANIZATION      "ParaView"
+  ORGANIZATION      "TTK"
   VERSION           "${PARAVIEW_VERSION_MAJOR}.${PARAVIEW_VERSION_MINOR}.${PARAVIEW_VERSION_PATCH}"
   ${style_args}
   MAIN_WINDOW_CLASS "ParaViewMainWindow"
