--- a/Qt/ApplicationComponents/pqExampleVisualizationsDialog.cxx	2018-04-16 15:17:15.587864128 +0200
+++ b/Qt/ApplicationComponents/pqExampleVisualizationsDialog.cxx	2018-04-16 15:52:55.271002381 +0200
@@ -19,13 +19,11 @@
   ui->setupUi(this);

   QObject::connect(
-    this->ui->CanExampleButton, SIGNAL(clicked(bool)), this, SLOT(onButtonPressed()));
+    this->ui->Example1Button, SIGNAL(clicked(bool)), this, SLOT(onButtonPressed()));
   QObject::connect(
-    this->ui->DiskOutRefExampleButton, SIGNAL(clicked(bool)), this, SLOT(onButtonPressed()));
+    this->ui->Example2Button, SIGNAL(clicked(bool)), this, SLOT(onButtonPressed()));
   QObject::connect(
-    this->ui->WaveletExampleButton, SIGNAL(clicked(bool)), this, SLOT(onButtonPressed()));
-  QObject::connect(
-    this->ui->HotGasAnalysisExampleButton, SIGNAL(clicked(bool)), this, SLOT(onButtonPressed()));
+    this->ui->Example3Button, SIGNAL(clicked(bool)), this, SLOT(onButtonPressed()));
 }

 //-----------------------------------------------------------------------------
@@ -43,26 +41,21 @@
   {
     const char* stateFile = NULL;
     bool needsData = false;
-    if (button == this->ui->CanExampleButton)
+    if (button == this->ui->Example1Button)
     {
-      stateFile = ":/pqApplicationComponents/ExampleVisualizations/CanExample.pvsm";
+      stateFile = ":/pqApplicationComponents/ExampleVisualizations/Example1.pvsm";
       needsData = true;
     }
-    else if (button == this->ui->DiskOutRefExampleButton)
+    else if (button == this->ui->Example2Button)
     {
-      stateFile = ":/pqApplicationComponents/ExampleVisualizations/DiskOutRefExample.pvsm";
+      stateFile = ":/pqApplicationComponents/ExampleVisualizations/Example2.pvsm";
       needsData = true;
     }
-    else if (button == this->ui->WaveletExampleButton)
+    else if (button == this->ui->Example3Button)
     {
-      stateFile = ":/pqApplicationComponents/ExampleVisualizations/WaveletExample.pvsm";
+      stateFile = ":/pqApplicationComponents/ExampleVisualizations/Example3.pvsm";
       needsData = false;
     }
-    else if (button == this->ui->HotGasAnalysisExampleButton)
-    {
-      stateFile = ":/pqApplicationComponents/ExampleVisualizations/HotGasAnalysisExample.pvsm";
-      needsData = true;
-    }
     else
     {
       qCritical("No example file for button");
