diff --git c/CMakeLists.txt w/CMakeLists.txt
index 792bfff6..3870fbb5 100644
--- c/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -803,3 +803,21 @@ install(FILES ${ParaView_SOURCE_DIR}/TTK/Data/Example2.vti
 install(FILES ${ParaView_SOURCE_DIR}/TTK/Data/Example3.vti
   DESTINATION share/paraview-5.8/examples/
   COMPONENT development)
+
+#-----------------------------------------------------------------------------
+# Generate a .deb package
+set(CPACK_PACKAGE_NAME "TTK-ParaView")
+set(CPACK_PACKAGE_FILE_NAME "ttk-paraview")
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "ParaView built with TTK patches")
+set(CPACK_PACKAGE_CONTACT "Julien Tierny <julien.tierny@sorbonne-universite.fr>")
+set(CPACK_PACKAGE_VENDOR "LIP6 - Sorbonne Université")
+set(CPACK_PACKAGE_HOMEPAGE_URL "https://topology-tool-kit.github.io/")
+set(CPACK_DEBIAN_PACKAGE_DEPENDS
+  "qt5-default, qttools5-dev, libqt5x11extras5-dev, qtxmlpatterns5-dev-tools, libqt5svg5-dev, libxt-dev"
+  )
+set(CPACK_PACKAGE_VERSION_MAJOR ${PARAVIEW_VERSION_MAJOR})
+set(CPACK_PACKAGE_VERSION_MINOR ${PARAVIEW_VERSION_MINOR})
+set(CPACK_PACKAGE_VERSION_PATCH ${PARAVIEW_VERSION_PATCH})
+# autogenerate dependency information
+set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS ON)
+include(CPack)
