--- a/Qt/ApplicationComponents/Resources/UI/pqExampleVisualizationsDialog.ui	2017-06-15 15:28:01.000000000 -0700
+++ b/Qt/ApplicationComponents/Resources/UI/pqExampleVisualizationsDialog.ui	2017-06-15 15:32:43.000000000 -0700
@@ -11,7 +11,7 @@
    </rect>
   </property>
   <property name="windowTitle">
-   <string>ParaView Example Visualizations</string>
+   <string>TTK Example Visualizations</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
@@ -25,7 +25,7 @@
       </font>
      </property>
      <property name="text">
-      <string>ParaView Example Visualizations</string>
+      <string>TTK Example Visualizations</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
@@ -55,7 +55,7 @@
           <number>2</number>
          </property>
          <item>
-          <widget class="QPushButton" name="CanExampleButton">
+          <widget class="QPushButton" name="Example1Button">
            <property name="enabled">
             <bool>true</bool>
            </property>
@@ -88,7 +88,7 @@
            </property>
            <property name="icon">
             <iconset>
-             <normalon>:/pqApplicationComponents/Thumbnails/CanExample.png</normalon>
+             <normalon>:/pqApplicationComponents/Thumbnails/Example1.png</normalon>
             </iconset>
            </property>
            <property name="iconSize">
@@ -109,7 +109,7 @@
           </widget>
          </item>
          <item>
-          <widget class="QLabel" name="CanExampleDescription">
+          <widget class="QLabel" name="Example1Description">
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
@@ -120,7 +120,7 @@
             <number>0</number>
            </property>
            <property name="text">
-            <string>Exodus II file with timesteps, Clip filter</string>
+            <string>Scalar data example</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
@@ -146,7 +146,7 @@
           <number>2</number>
          </property>
          <item>
-          <widget class="QPushButton" name="DiskOutRefExampleButton">
+          <widget class="QPushButton" name="Example2Button">
            <property name="minimumSize">
             <size>
              <width>300</width>
@@ -158,7 +158,7 @@
            </property>
            <property name="icon">
             <iconset>
-             <normalon>:/pqApplicationComponents/Thumbnails/DiskOutRefExample.png</normalon>
+             <normalon>:/pqApplicationComponents/Thumbnails/Example2.png</normalon>
             </iconset>
            </property>
            <property name="iconSize">
@@ -173,9 +173,9 @@
           </widget>
          </item>
          <item>
-          <widget class="QLabel" name="DiskOutRefExampleDescription">
+          <widget class="QLabel" name="Example2Description">
            <property name="text">
-            <string>Exodus II file, Clip filter, Stream Tracer filter, Tube filter, Glyph filter</string>
+            <string>Bivariate scalar data example</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
@@ -195,7 +195,7 @@
           <number>2</number>
          </property>
          <item>
-          <widget class="QPushButton" name="WaveletExampleButton">
+          <widget class="QPushButton" name="Example3Button">
            <property name="minimumSize">
             <size>
              <width>300</width>
@@ -207,7 +207,7 @@
            </property>
            <property name="icon">
             <iconset resource="../pqApplicationComponents.qrc">
-             <normaloff>:/pqApplicationComponents/Thumbnails/WaveletExample.png</normaloff>:/pqApplicationComponents/Thumbnails/WaveletExample.png</iconset>
+             <normaloff>:/pqApplicationComponents/Thumbnails/Example3.png</normaloff>:/pqApplicationComponents/Thumbnails/Example3.png</iconset>
            </property>
            <property name="iconSize">
             <size>
@@ -221,12 +221,12 @@
           </widget>
          </item>
          <item>
-          <widget class="QLabel" name="WaveletExampleDescription">
+          <widget class="QLabel" name="Example3Description">
            <property name="mouseTracking">
             <bool>false</bool>
            </property>
            <property name="text">
-            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Wavelet with volume rendering and contours&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
+            <string>Uncertain scalar data example</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
