--- a/ParaViewCore/VTKExtensions/Default/vtkPVSelectionSource.cxx	2016-01-08 21:21:35.000000000 +0100
+++ b/ParaViewCore/VTKExtensions/Default/vtkPVSelectionSource.cxx	2016-06-06 17:50:25.601307781 +0200
@@ -138,7 +138,7 @@
     };

   typedef std::set<vtkIdType> SetOfIDs;
-  typedef std::set<IDType> SetOfIDType;
+  typedef std::vector<IDType> SetOfIDType;
   typedef std::set<CompositeIDType> SetOfCompositeIDType;
   typedef std::set<HierarchicalIDType> SetOfHierarchicalIDType;
   typedef std::set<PedigreeIDType> SetOfPedigreeIDType;
@@ -248,7 +248,18 @@
     piece = -1;
     }
   this->Mode = ID;
-  this->Internal->IDs.insert(vtkInternal::IDType(piece, id));
+
+  bool inThere = false;
+  for(int i = 0; i < (int) this->Internal->IDs.size(); i++){
+    if(this->Internal->IDs[i].ID == vtkInternal::IDType(piece, id).ID){
+      inThere = true;
+      break;
+    }
+  }
+  if(!inThere){
+    // no duplicated entry
+    this->Internal->IDs.push_back(vtkInternal::IDType(piece, id));
+  }
   this->Modified();
 }
